src/client/pages/Offer/Introduction/DetailsModal/index.tsx

```diff
@@ -0,0 +1,227 @@
+import { Modal, ModalProps } from 'components/ModalNew'
+import {
+  BundledQuote,
+  useCreateQuoteBundleMutation,
+  useQuoteCartQuery,
+  QuoteBundleVariant,
+} from 'data/graphql'
+
+import { useTextKeys } from 'utils/textKeys'
+import { useCurrentLocale } from 'l10n/useCurrentLocale'
+import { getBundleVariantFromInsuranceTypes } from 'pages/OfferNew/utils'
+import { useSelectedInsuranceTypes } from 'utils/hooks/useSelectedInsuranceTypes'
+import { QuoteInput } from './types'
+import { Details, getValidationSchema } from './Details'
+
+const Container = styled.div`
+  width: 100%;
+  height: 100%;
+  padding: 1rem 2rem;
+  display: flex;
+  flex-direction: column;
+
+  @media (max-width: 900px) {
+    padding: 3rem 3rem 1.5rem 3rem;
+  }
+
+  @media (max-width: 800px) {
+    padding: 2rem 2rem 1.5rem 2rem;
+  }
+
+  @media (max-width: 500px) {
+    padding: 2rem 1rem 1.5rem 1rem;
+  }
+`
+
+const Headline = styled.div`
+  font-family: ${fonts.FAVORIT};
+  font-size: 2.5rem;
+  line-height: 3.5rem;
+  color: ${colorsV3.black};
+
+  @media (max-width: 600px) {
+    font-size: 2rem;
+    line-height: 3rem;
+  }
+`
+
+const Footer = styled.div`
+  padding: 1rem 0;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+`
+
+const Warning = styled.div`
+  font-size: 0.75rem;
+  line-height: 1rem;
+  margin-top: 1rem;
+  color: ${colorsV3.gray900};
+  text-align: center;
+`
+
+const ErrorMessage = styled.div`
+  font-size: 0.75rem;
+  line-height: 1rem;
+  margin-top: 1rem;
+  color: ${colorsV3.red500};
+  text-align: center;
+`
+
+const LoadingDimmer = styled.div<{ visible: boolean }>`
+  background: rgba(255, 255, 255, 0.5);
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  transition: all 350ms;
+  opacity: ${(props) => (props.visible ? 1 : 0)};
+  visibility: ${(props) => (props.visible ? 'visible' : 'hidden')};
+  z-index: 2;
+`
+
+type DetailsModalProps = {
+  quoteCartId: string
+  allQuotes: BundledQuote[]
+  refetch: () => Promise<void>
+}
+
+export const DetailsModal: React.FC<ModalProps & DetailsModalProps> = ({
+  quoteCartId,
+  refetch,
+  allQuotes,
+  isVisible,
+  onClose,
+}) => {
+  const textKeys = useTextKeys()
+  const { isoLocale, marketLabel } = useCurrentLocale()
+
+  const [
+    createQuoteBundle,
+    {
+      data: createQuoteBundleData,
+      error: createQuoteBundleError,
+      loading: isBundleCreationInProgress,
+    },
+  ] = useCreateQuoteBundleMutation()
+
+  const { data } = useQuoteCartQuery({
+    variables: { id: quoteCartId, locale: isoLocale },
+  })
+  const [selectedInsuranceTypes] = useSelectedInsuranceTypes()
+
+  const bundleVariants = (data?.quoteCart.bundle?.possibleVariations ??
+    []) as Array<QuoteBundleVariant>
+  const selectedQuoteBundle =
+    getBundleVariantFromInsuranceTypes(
+      bundleVariants,
+      selectedInsuranceTypes,
+    ) || bundleVariants[0]
+
+  if (!selectedQuoteBundle) return null
```

Comment:
Is this representing a loading state? Not great but I think we can live with it

Reply:
`selectedQuoteBundle` should be available as the page is loaded, so it's not a loading state. It's because `getBundleVariantFromInsuranceTypes` can potentially return null which should never be the case, and if it is we should handle it much earlier than here I believe.